internal/compilationcache/compilationcache.go

```diff
@@ -0,0 +1,44 @@
+package compilationcache
+
+import (
+	"crypto/sha256"
+	"io"
+)
+
+// Cache is the interface for compilation caches. Internally, the cache
+// here means that the compiled binary cache by compilers. Regardless of the usage of
+// ExternCache, the compiled functions are cached in memory, but its lifetime is
+// bound to the lifetime of wazero.Runtime or wazero.CompiledModule.
+// Usually, the compilation of Wasm binary is time-consuming. Therefore, you might
+// want to cache the compilation result across the processes of wazero users.
+//
+// See NewFileCache for the example implementation.
+type Cache interface {
+	// Get is called when the runtime is trying to get the cached content.
+	// Implementations are supposed to return `content` which can be used to
```

Comment:
this is too wordy, maybe refer to map for an example of how this works? since content is only used for compilation of functions seems we can be more concrete generally